org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/references/participants/XMLReferencesDiagnosticParticipant.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.references.participants;
+
+import java.text.MessageFormat;
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.extensions.contentmodel.settings.XMLValidationSettings;
+import org.eclipse.lemminx.extensions.references.XMLReferencesPlugin;
+import org.eclipse.lemminx.extensions.references.search.ReferenceLink;
+import org.eclipse.lemminx.extensions.references.search.SearchEngine;
+import org.eclipse.lemminx.extensions.references.search.SearchNode;
+import org.eclipse.lemminx.extensions.references.search.SearchNode.ValidationStatus;
+import org.eclipse.lemminx.services.extensions.diagnostics.IDiagnosticsParticipant;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+/**
+ * XML references diagnostics which reports undefined references.
+ *
+ */
+public class XMLReferencesDiagnosticParticipant implements IDiagnosticsParticipant {
+
+	private final static String UNDEFINED_REFERENCE_MESSSAGE = "Undefined reference ''{0}'': nothing that matches the expression ''{1}'' defines ''{2}''.";
+
+	private final static String INVALID_PREFIX_MESSSAGE = "Invalid prefix ''{0}'': that matches the expression ''{1}'' requires the ''{2}'' prefix.";

```

Comment:
I think it might be more clear if it said:

"Invalid reference ''{0}'': references that match the expression ''{1}'' require the ''{2}'' prefix"